src/vcpkg/commands.applocal.cpp

```diff
@@ -415,17 +415,34 @@ namespace
             const auto source = installed_dir / target_binary_name;
             const auto target = target_binary_dir / target_binary_name;
 
-            const auto mutant_name = Strings::to_utf16("vcpkg-applocal-" + Hash::get_string_sha256(target_binary_dir));
-
-            HANDLE the_mutant = ::CreateMutexW(nullptr, FALSE, mutant_name.c_str());
-
-            if (the_mutant == NULL)
+            const auto mutant_name = "vcpkg-applocal-" + Hash::get_string_sha256(target_binary_dir);
+            struct MutantGuard
+            {
+                MutantGuard(StringView name)
+                {
+                    h = ::CreateMutexW(nullptr, FALSE, Strings::to_utf16(name).c_str());
+                    if (h)
+                    {
+                        WaitForSingleObject(h, INFINITE);
+                    }
+                }
+                explicit operator bool() const { return h; }
+                ~MutantGuard()
+                {
+                    if (h)
+                    {
+                        ReleaseMutex(h);
+                        CloseHandle(h);
+                    }
+                }
+                HANDLE h;
+            };
+            const MutantGuard mutant(mutant_name);
+            if (!mutant)
             {
-                Checks::exit_with_message(VCPKG_LINE_INFO, "Failed\n");
+                Checks::exit_with_message(VCPKG_LINE_INFO, "Failed to acquire mutant %s", mutant_name);
             }
```

Comment:
IMO it would be better to just make the ctor do this which also lets you get rid of the operator bool et al.